### PR TITLE
[ENH] allow using of mkdoc admonitions

### PR DESCRIPTION
**Pull request by Remi-Gau**
- add a script to remove mkdoc admonitions 
- run it during pdf build
- add minimal testing to help debugging if needed
- add example admonition to highlight the bids examples repo

### CODE REVIEW

File: pdf_build_src/tests/data/expected/README.md
Comment (Remi-Gau) [inline pdf_build_src/tests/data/expected/README.md:1]: TODO: add proper tests to check that we get the expected output

File: pdf_build_src/process_markdowns.py [hunk 1]
```diff
@@ -679,33 +681,38 @@ def process_macros(duplicated_src_dir_path):
 
     duplicated_src_dir_path = "src_copy/src"
 
-    # Step 1: make a copy of the src directory in the current directory
+    # make a copy of the src directory in the current directory
```
Comment (Remi-Gau) [inline pdf_build_src/process_markdowns.py:684]: Removed the step ordering... Because we will have to change it everytime steps are added.

**Comment by Remi-Gau** [comment]
Example of admonition:

https://bids-specification--1711.org.readthedocs.build/en/1711/modality-specific-files/near-infrared-spectroscopy.html

image

[Rendered PDF](https://output.circle-artifacts.com/output/job/93c76474-35a4-4d5f-953c-7a95b3fe7372/artifacts/0/bids-spec.pdf):

image

**Comment by Remi-Gau** [comment]
@tsalo @sappelhoff @effigies 
This is a draft but as far as I can tell this would allow using admonitions without breaking the PDF build.

Can you have a quick look and tell me if you see something I am missing?

**Comment by Remi-Gau** [comment]
the remark linting will be annoying though

**Review by sappelhoff — COMMENTED** [review]
Great! Thanks a lot @Remi-Gau, I think this looks promising and can be deployed.

> the remark linting will be annoying though

an easy fix would be to switch off that particular linting check, which I think would be fine.

**Comment by Remi-Gau** [comment]
> an easy fix

OK so it seems that [remark-lint-code-block-style](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-code-block-style) is set to `fenced` by [remark-preset-lint-markdown-style-guide
](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-preset-lint-markdown-style-guide)

So need to find how to unset that.

**Comment by tsalo** [comment]
This looks awesome!

File: pdf_build_src/test_remove_admonitions.py
Comment (Remi-Gau) [inline pdf_build_src/test_remove_admonitions.py:1]: Note that this is not run automatically in CI.

**Comment by Remi-Gau** [comment]
PDF looks fine to me.

Note that some links are not rendered in the pdf but this is not related to this PR.

For example the link at the top of the genetic section:

````
[`UK biobank`](https://github.com/bids-standard/bids-examples/tree/master/genetics_ukbb)
````

I think this is because of the back tick in the square brackets.

**Comment by Remi-Gau** [comment]
For contributors: should we preface our admonitions with some markdown comments the way we do it the macros so that users are not confused about what they are ?

**Comment by tsalo** [comment]
I don't think that's necessary. Macros aren't exactly common, but admonitions are. We can have that info in a style guide if we want though.

**Comment by Remi-Gau** [comment]
> I don't think that's necessary. Macros aren't exactly common, but admonitions are. We can have that info in a style guide if we want though.

OK so something to add in the contributing.md

**Review by sappelhoff — APPROVED** [review]

**Comment by Remi-Gau** [comment]
Actually before merging.

- handle admonition that start with ??? and ???+
- possibly simplify mkdoc macro that adds admonition below the filename templates

**Review by ericearl — APPROVED** [review]
lgtm!

**Pull request merged**